core_impl/src/test/java/io/opencensus/stats/StatsTestUtil.java

```diff
@@ -80,6 +81,93 @@ private static StatsContextImpl createContext(
   }
 
   /**
+   * Compare the actual and expected AggregationMap within the given tolerance.
+   *
+   * @param expected the expected map.
+   * @param actual the actual mapping from {@code List<TagValue>} to
+   *     {@code List<AggregationData>}.
+   * @param tolerance the tolerance used for {@code double} comparison.
+   * @throws AssertionError if the AggregationMap does not contain exactly the keys and values.
+   */
+  static void assertAggregationMapEquals(
+      Map<List<TagValue>, List<AggregationData>> actual,
+      Map<List<TagValue>, List<AggregationData>> expected,
+      double tolerance) {
+    assertThat(actual.keySet()).containsExactlyElementsIn(expected.keySet());
+    for (List<TagValue> tagValues : actual.keySet()) {
+      assertAggregationDataListEquals(expected.get(tagValues), actual.get(tagValues), tolerance);
+    }
+  }
+
+  /**
+   * Compare the expected and actual list of {@code AggregationData} within the given tolerance.
+   *
+   * @param expectedValue the expected list of {@code AggregationData}.
+   * @param actualValue the actual list of {@code AggregationData}.
+   * @param tolerance the tolerance used for {@code double} comparison.
+   * @throws AssertionError if the AggregationMap does not contain exactly the keys and values.
+   */
+  static void assertAggregationDataListEquals(
+      List<AggregationData> expectedValue, List<AggregationData> actualValue,
+      final double tolerance) {
+    assertThat(expectedValue.size()).isEqualTo(actualValue.size());
+    for (int i = 0; i < expectedValue.size(); i++) {
+      AggregationData actual = actualValue.get(i);
+      final AggregationData expected = expectedValue.get(i);
+      actual.match(
```

Comment:
Correct, since `assertAggregationDataEquals` is not being used anywhere else, I've removed that method and update the validation here.